testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1174,3 +1185,65 @@ fn test_malformed_script() {
         .mint_coins(&["mintb", "0", "10"], true)
         .unwrap();
 }
+
+#[test]
+fn test_key_manager_consensus_rotation() {
+    // Create and launch a local validator swarm of 2 nodes.
+    let mut swarm = TestEnvironment::new(2);
+    swarm.launch_swarm(RoleType::Validator);
+
+    // Create a node config for the key manager by modifying the first node config in the swarm.
+    let config_path = swarm.validator_swarm.config.config_files.get(0).unwrap();
+    let mut config = NodeConfig::load(&config_path).unwrap();
+    let json_rpc_endpoint = format!("http://127.0.0.1:{}", config.rpc.address.port());
+    let backend = config.consensus.safety_rules.backend.clone();
+    config.secure.key_manager.json_rpc_endpoint = json_rpc_endpoint.clone();
+    config.secure.key_manager.secure_backend = backend.clone();
+    config.secure.key_manager.rotation_period_secs = 15;
+    config.secure.key_manager.sleep_period_secs = 15;
+    config.save(config_path).unwrap();
+
+    // Bootstrap secure storage by initializing the keys required by the key manager.
+    // TODO(joshlind): FIX ME -- these keys should already be set when the validator is initialized.
+    let mut storage: Box<dyn Storage> = (&backend).try_into().unwrap();
+    storage
+        .create_key("consensus_previous", &Policy::default())
+        .unwrap();
+    let operator_private = config
+        .test
+        .unwrap()
+        .operator_keypair
+        .unwrap()
+        .take_private()
+        .unwrap();
+    storage
+        .create_with_default_policy(OPERATOR_KEY, Value::Ed25519PrivateKey(operator_private))
+        .unwrap();
+
+    // Create a json-rpc connection to the blockchain and verify storage matches the on-chain state.
+    let libra_interface = JsonRpcLibraInterface::new(json_rpc_endpoint);
+    let account = config.validator_network.clone().unwrap().peer_id;
+    let current_consensus = storage.get_public_key(CONSENSUS_KEY).unwrap().public_key;
+    let validator_info = libra_interface.retrieve_validator_info(account).unwrap();
+    assert_eq!(&current_consensus, validator_info.consensus_public_key());
+
+    // Spawn the key manager and sleep until a rotation occurs.
+    let mut command = Command::new(workspace_builder::get_bin(KEY_MANAGER_BIN));
+    command
+        .current_dir(workspace_builder::workspace_root())
+        .arg(config_path)
+        .stdout(Stdio::null())
```

Comment:
I've just piped the key manager to /dev/null for now.. It probably makes sense to create a log file for the key manager..

Reply:
this is going to be super tricky, do you have a strategy? could you add a todo?

Reply:
i thought the plan here was to just push this out to the standard logger attached to this iff logging was enabled for the test?

Reply:
Hmm.. it's not super obvious to me right now, so I've added a TODO and piped the output to /dev/null (so it doesn't confuse anyone running the tests).

Reply:
@davidiw FYI: this turned out not to be super straight forward. The issue is that the smoke tests don't currently support logging (i.e., the smoke tests are just standard rust tests, so they only handle printing to stdout/stderr in the case of a test failure -- they don't have any special logger attached to them). To overcome this, I've piped the key manager output to the test process (which, will simply print the key manager output to the console on a failure). This way, on successful runs, the output is not printed, but on failures, we can debug the key manager. This seems acceptable enough for now (that is, until we implement dedicated key manager logs).